src/Rule/Uuid.php

```diff
@@ -23,17 +21,41 @@ class Uuid extends Regex
     const INVALID_UUID = 'Uuid::INVALID_UUID';
 
     /**
-     * UUID Version 4.
+     * UUID NIL & version binary masks
      */
-    const UUID_V4 = 4;
+    const UUID_NIL = 0b000001;
+    const UUID_V1  = 0b000010;
+    const UUID_V2  = 0b000100;
+    const UUID_V3  = 0b001000;
+    const UUID_V4  = 0b010000;
```

Comment:
The thing is that these are essentially ints right now. To do this and keep BC, you'd have to map `0b000100` to V4 (thus 4=4). That'd keep BC but make the mapping a bit illogical.

Right now the given value has no meaning so while it is officially a BC break, it's not really a big one. People using the constant won't break, only those that actively chose to input the number 4. But introducing the illogical mapping now can never really be changed anymore, breaking that BC would be big as the number is actively used (which it isn't now, as there's no other accepted input).

So the options are:
- Small BC break now with limited results
- Illogical bit order due to keeping BC, impossible to make logical later because that BC break would be big
- A non-bitmask solution which can work with an array of numbers for example